api/src/main/java/com/github/eyefloaters/console/api/model/PartitionKey.java

```diff
@@ -0,0 +1,20 @@
+package com.github.eyefloaters.console.api.model;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.common.TopicPartition;
+
+public record PartitionKey(String topicId, String topicName, int partition) {
```

Comment:
My mind was in the RDBMS world when naming this. I.e. primary key describing the columns, but the actual tuple being the ID. I'm fine with calling it an ID, but do we really need to prefix it with `Topic` or would just `PartitionId` be fine? At this point topics are the only thing being partitioned and it seems pretty established as a stand-alone term in the Kafka domain.

Regarding the nullability, when an API client sees an instance of this it should always have both ID and name. Internally, however, the ID is nullable since this is used as a bridge between some operations that use topic ID, and others that don't support it. I'll add a doc about that being the case.

Reply:
I've renamed `PartitionKey` to `PartitionId`, made all fields non-nullable, and adjusted how it is initially constructed when loading the consumer groups. Please comment back or let me know otherwise your thoughts on using `PartitionId` if you think it could be problematic longer term.